warehouse/query-core/src/main/java/datawave/query/jexl/visitors/ExpandMultiNormalizedTerms.java

```diff
@@ -155,12 +155,9 @@ public Object visit(ASTReference node, Object data) {
      * @return
      */
     protected boolean isDelayedPredicate(JexlNode currNode) {
-        if (ASTDelayedPredicate.instanceOf(currNode) || ExceededOrThresholdMarkerJexlNode.instanceOf(currNode)
+        return ASTDelayedPredicate.instanceOf(currNode) || ExceededOrThresholdMarkerJexlNode.instanceOf(currNode)
                         || ExceededValueThresholdMarkerJexlNode.instanceOf(currNode) || ExceededTermThresholdMarkerJexlNode.instanceOf(currNode)
-                        || IndexHoleMarkerJexlNode.instanceOf(currNode) || ASTEvaluationOnly.instanceOf(currNode))
-            return true;
-        else
-            return false;
+                        || IndexHoleMarkerJexlNode.instanceOf(currNode) || ASTEvaluationOnly.instanceOf(currNode);
```

Comment:
```
        return QueryPropertyMarkerVisitor.instanceOf(currNode, Arrays.asList(ASTDelayedPredicate.class, ExceededOrThresholdMarkerJexlNode.class,
                        ExceededValueThresholdMarkerJexlNode.class, ExceededTermThresholdMarkerJexlNode.class, IndexHoleMarkerJexlNode.class,
                        ASTEvaluationOnly.class), null);
```

This method is a dupe of (erroneously protected) JexlASTHelper.isDelayedPredicate.  We should rip this out, update the JexlASTHelper method to be public, and use that instead. 

Also, instead of visiting the tree six times for each instanceof call, we should pass the list of types to the QueryPropertyMarkerVisitor, and run instanceof once (as shown above);

Reply:
I've deduped the method and used QueryPropertyMarkerVisitor. One change I made was to use QueryPropertyMarkerVisitor.instanceOfAny(node) instead, since that will default to the TYPE_IDENTIFIERS set that already contains the delayed predicate instance types.

Reply:
Good call.  I didn't realize that there were only 6 marker nodes.  D'oh!  Nice work!